src/validation.cpp

```diff
@@ -2909,13 +2909,10 @@ bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool f
         if (!block.IsAuxpow())
             return error("%s : auxpow on block with non-auxpow version", __func__);
 
-
-        // Can't do these since current aux chain id is not known.
-        //if (!block.auxpow->check(block.GetHash(), block.GetChainId(), params))
-        //    return error("%s : AUX POW is not valid", __func__);
-        //if (!CheckProofOfWork(block.auxpow->getParentBlockPoWHash(), block.nBits, params))
-        //    return error("%s : AUX proof of work failed", __func__);
-        //
+        if (!CheckProofOfWork(block.auxpow->getParentBlockPoWHash(), block.nBits, params))
+            return error("%s : AUX proof of work failed", __func__);
+        if (!block.auxpow->check(block.GetHash(), block.GetChainId(), params))
+            return error("%s : AUX POW is not valid", __func__);
```

Comment:
I think we should save this for a different PR given the nature of the intended fix.